### DISCUSSION TITLE
Gitpod: URL changes sometimes which may require changing .env

### DISCUSSION THREAD

**Discussion by ngillux** [freeCodeCamp/classroom#332] [General]
If using Gitpod this may be useful to you.

In previous Gitpod session(s) my URL was https://3000-XXXX-XXXXX.ws-us99.gitpod.io/, however, when going into a session today it had changed to  https://3000-XXXX-XXXXX.ws-us100.gitpod.io/.

This resulted in my NEXT_AUTH_URL and MOCK_JSON_SERVER_URL not being hit, since I had set those to be https://3000-XXXX-XXXXX.ws-us99.gitpod.io/ [*note the 'us99'].

To resolve this I updated the URL to contain 'us100' in my .env file, and also updated the OAuth app in GitHub Developer Settings (just modified the 'us99' to be 'us100'] for the respective URLs (NEXT_AUTH_URL and MOCK_JSON_SERVER_URL).

Curious if anyone else has ran into a similar issue.

**Comment by GuillermoFloresV** [comment]
Hi, @ngillux This will happen regularly, and as far as I know, there is no "quick fix" for it. GitPod regularly changes URLs due to many possible reasons:

<details><summary>Server Migration</summary>
  <p>This can be caused by behind-the-scenes GitPod infrastructure migration which can cause new URLs due to different naming conventions or simply the workspace now living on a different host server
  </p>
</details> 
<hr>
<details><summary>Workspace Updates</summary>
  <p>
  Workspace updates made by GitPod (these happen very regularly) could it to move to a new server, where its new configuration is being used, hence the need to rebuild the workspace and the URL (since the old one was destroyed when moving to a new server)
  </p>
</details> 
<hr>
<details><summary>Workspace Inactivity</summary>
  <p>
  GitPod workspaces are inherently temporary. Because of this, they are deleted after some time of inactivity to free up resources, when the workspace is deleted, the URL will no longer work.
  </p>
</details> 

I have looked around for some possible solutions to this issue if you feel it is worth exploring to make your development experience slightly less annoying.

1. [Local companion for GitPod](https://www.gitpod.io/blog/local-app), This solution runs the app on your computer, rather than on GitPod, and creates a secure connection to your workspace allowing you to use `localhost` as your "stable URL"
2. [Cloudflared tunnel](https://developers.cloudflare.com/cloudflare-one/tutorials/share-new-site/#create-dns-records) As far as I can tell, this is a similar solution, but instead you forward through a Cloudflare service which acts as your "stable URL".

I have not personally tried to use either of these solutions, but if you'd like we could try to go over them during a work session either the regularly scheduled one or one outside of that scheduled time. Let me know what you think or if you have any questions, feel free to tag me to make sure I get a notification :)

**Reply by ngillux** [reply]
Hi @GuillermoFloresV  thanks for providing a clarification and context, as well as possible solutions! I think I've gotten used to it now, but I'd definitely like to explore option 1.

Maybe we can do this in the regularly scheduled work session, but if you have time outside, please let me know and I'd be happy to accommodate your availability.

Thanks!
